grails-app/controllers/org/pih/warehouse/api/PurchaseOrderApiController.groovy

```diff
@@ -155,7 +156,7 @@ class PurchaseOrderApiController {
                 order?.orderedOrderItems?.size() ?: 0,
                 order?.shippedOrderItems?.size() ?: 0,
                 order?.receivedOrderItems?.size() ?: 0,
-                order?.invoiceItems?.size() ?: 0,
+                invoiceService.countInvoicedOrderItems(order)?:0,
```

Comment:
`order?.invoiceItems?.size()` should not be used here at all in the beginning, because it's taking all invoice items instead of invoiced order items, plus I even left a comment above `order?.invoiceItems` some time ago that it should be used for one order only but not for a list of orders like it was used in this case.

Perhaps it would be better to add all these counts to the OrderSummary? We would have then everything in one place without the need to have additional queries. All `orderedOrderItems`, `shippedOrderItems`, `receivedOrderItems`, and `invoicedOrderItems` could be additionally (easily) calculated in the `helper views` and stored in the OrderSummary.

Refactoring or improving `orderItem?.invoiceItems` and `orderItem?.quantityInvoicedInStandardUom` should also be done definitely but it probably doesn't have to be done for PO export if were getting the item counts from view or the query that you created.